vm/thread.go

```diff
@@ -204,12 +204,12 @@ func (t *thread) evalMethodObject(receiver Object, method *MethodObject, receive
 	argTypesCount := len(method.argTypes())
 
 	for _, at := range method.argTypes() {
-		if at == bytecode.NormalArg {
+		if at == bytecode.NormalArg || at == bytecode.RequiredKeywordArg {
 			minimumArgNumber++
 		}
 	}
 
-	if argC > method.argc && method.lastArgType() != bytecode.SplatArg {
+	if argC > method.argc && !method.isSplatArgIncluded() {
```

Comment:
Maybe we should start a new issue to write down a table for all the possible cases and checks, so we can have a bigger picture about the argument checking. And that can also be the future documentation as well